### ISSUE TITLE
Nancy.ViewEngines.Razor has the wrong dependency for Microsoft.AspNet.Razor

### ISSUE THREAD

**Issue by KnowMoreIT** [NancyFx/Nancy#2582]
### Description

The project.json file is targeting the wrong build of Razor. It is configured to install >=2.0.30506.0 but it should be statically set to require =2.0.30506.0 exactly.

I "believe" you can change this file and line: https://github.com/NancyFx/Nancy/blob/master/src/Nancy.ViewEngines.Razor/project.json#L15 to "Microsoft.AspNet.Razor": "[2.0.30506.0]", 

which will only allow 2.0.30506.0

**Comment by thecodejunkie** [comment]
I just tried installing `Nancy.ViewEngines.Razor` version `2.0.0-barneyrubble` in both `packages.config` and `project.json` and they all resolved fine. Maybe there are some issues in your dependency tree which is causing it to bump the version number?

### Using project.json
![image](https://cloud.githubusercontent.com/assets/50543/19223924/143a081c-8e7b-11e6-80da-2afd322b2aa2.png)

### Using packages.config
![image](https://cloud.githubusercontent.com/assets/50543/19223928/1eab5dc8-8e7b-11e6-9d55-ef570962a180.png)

According to @phillip-haydon there should not be any issues with going to v3 as he did a lot of testing around it before

**Comment by KnowMoreIT** [comment]
My .NET is actually configured for 4.6 but we were talking about this in the chat with nuget specifying >=[Version] instead of just the specific version. It did seem to work with the later version except for when trying to publish the project (precompiling). It could have been another conflict but I was thinking you were saying the specific version should of been 2.0.30506. 

After removing everything and installing the depencies manually first (2.0.30506) I was able to get it to work just fine.

**Comment by phillip-haydon** [comment]
@KnowMoreIT we support both V2 and V3.

When I made the change we had users who wanted to run both MVC and Nancy in the same app for various reasons, which meant running V3 razor to support MVC at the same time.

I spent a lot of time looking at the changes in V3 of razor and at the time concluded that they were only bug fixes and .NET version bump (4.5 from 4.0)

By default we install V2 but if you run updates it can pick V3, it should still work though. 

I'm running a legacy project at home with 4.6.2 (and have upgraded from 4.5.0/1/2, 4.6/4.6.1) without any issues.

I have not tested in a core project however, but if you're still targeting .NET 4.6 then it should be working fine.

**Comment by thecodejunkie** [comment]
@phillip-haydon can you confirm that we can close this?

**Comment by phillip-haydon** [comment]
Ok I tested this before bed last night. 

Created a .NET 4.5, 4.5.2, and 4.6.2 projects, all pulling Nancy Razor.

They pulled in V2 of Razor by default. 

This works fine.

I then updated to the latest 3.2.3 version of razor. And it continued to work fine.

I then tried uploading this to my AWS Server before terminating my account (free tier gone 😢) so I tried just compiling, then using the publish feature. Both worked as expected for V2 and V3. (I only tested 4.6.2 as I was too lazy to do it for the other versions)

-----

I did not test on a core project.

I'm fine to close this @thecodejunkie I don't believe there is any issue.

-----

It is worth noting that we don't support pre-compilation of razor views.
